src/stdlib_linalg.fypp

```diff
@@ -80,8 +88,93 @@ module stdlib_linalg
     #:endfor
   end interface outer_product
 
+
+  ! Check for squareness
+  interface is_square
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is square
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_4))
```

Comment:
Yes, the tests are quite unwieldy, which is why I originally inquired about using `fypp` or similar for testing, and I believe you or somebody else kindly pointed me to #494. I'm happy to separate them out however you'd like.

I'm not sure about what you mean by "a more specific link". As you noted, I've just been following the general structure inside the `stdlib_linalg` module, but if there is another module whose style/structure you'd like me to use that has more robust links etc., please let me know.

Reply:
> Can we have a more specific link here instead of the forth description? Same for all other links to the specs below.

Before merging, I usually check the links, and update them if needed (I am currently doing it for #272; I am linking to the main title of each procedure). I can have a check for this PR when I am finished with #272 to help @ghbrown

Reply:
I'll have to rebase #494 once this is merged. Chances are good that I have to deal with a 2000 line merge conflict for this file. Not impossible to handle, just a bit more tedious.

As for the links, there are somewhat more specific links available like https://stdlib.fortran-lang.org/page/specs/stdlib_linalg.html#eye-construct-the-identity-matrix for the `eye` function. If the synopsis is not included in the title, the link will also become much shorter.

Reply:
Gotcha, I'll start working on these. Thanks.

Reply:
A recent commit makes the change to more specific URLs. Please let me know if this what you intended, or if I've misunderstood something. Thanks.